### PR TITLE
build: Use system NSS when possible

### LINKED ISSUE #1711: Use system-installed NSS dependeny
Rather than checking out and building our own copy of NSS during the build of neqo, we should detect and use a system-installed NSS. This will require some heavy surgery to `neqo-crypto/build.rs` but should really simplify things.

### PR DESCRIPTION
**Pull request by larseggert**
When `NSS_DIR` is **not** set, we now attempt to use the system NSS, and check whether it is recent enough for us. The current NSS minimum version is in `neqo-crypto/min_version.txt`, so it can be accessed by Rust (via `neqo-crypto/src/min_version.rs`) and by the CI scripts.

Fixes #1711

### CODE REVIEW

File: neqo-crypto/build.rs
Comment (larseggert) [reply neqo-crypto/build.rs:190]: We could make it contingent on whether `NSS_DIR` is defined. (If it is, build and use that one.)

File: neqo-crypto/build.rs
Comment (larseggert) [reply neqo-crypto/build.rs:100]: Good suggestions, will look into it. First trying to get CI building again.

File: .github/actions/nss/action.yml
Comment (martinthomson) [inline .github/actions/nss/action.yml:30]:
```suggestion
        NSS_VERSION="$(pkg-config --modversion nss)"
```

File: .github/actions/nss/action.yml
Comment (martinthomson) [inline .github/actions/nss/action.yml:35]: So there is also `read NSS_MAJOR NSS_MINOR NSS_PATCH < <(echo "${NSS_VERSION//./ }")`.  ¯\\\_(ツ)\_/¯

File: neqo-crypto/build.rs
Comment (martinthomson) [inline neqo-crypto/build.rs:311]: Rather than split, because you don't need the slices, you can do this:

```suggestion
    let modversion_for_cmp = if modversion.chars().filter(|c| c == '.').count() == 1 {
```

File: neqo-crypto/build.rs
Comment (martinthomson) [inline neqo-crypto/build.rs:314]: You shouldn't need to clone here.  You already have an owned copy (that's what the `to_string()` above did.  That said, I would not use `to_string()` above, instead making a copy here instead.

File: neqo-crypto/build.rs
Comment (martinthomson) [inline neqo-crypto/build.rs:462]:
```suggestion
    } else if let Ok(nss_dir) = env::var("NSS_DIR") {
        setup_standalone(nss_dir)
    } else {
        pkg_config()
```

**Review by martinthomson — COMMENTED** [review]
I think that you can cut most of `fn nss_dir()` now that you won't be hitting it as often.  Just pass the env variable in to `setup_standalone()`

**Comment by larseggert** [comment]
> I think that you can cut most of `fn nss_dir()` now that you won't be hitting it as often. Just pass the env variable in to `setup_standalone()`

`nss_dir()` has the code that checks out NSS and NSPR if there is no copy already. Are you saying that should be cut, i.e., will we require people to do their own checkouts?

File: neqo-crypto/build.rs
Comment (martinthomson) [inline neqo-crypto/build.rs:318]:
```suggestion
    let version_req = VersionReq::parse(&format!(">={}", MINIMUM_NSS_VERSION.trim())).unwrap();
```

File: neqo-crypto/build.rs
Comment (martinthomson) [inline neqo-crypto/build.rs:18]: I think that you can use `mod min_version; use min_version::MINIMUM_NSS_VERSION;` rather than relying on `include!()`.

File: neqo-crypto/src/lib.rs
Comment (martinthomson) [inline neqo-crypto/src/lib.rs:62]: Again, I think that it is better to use:

```rust
#[path = "../min_version.rs"]
mod min_version;
use min_version::MINIMUM_NSS_VERSION;
```

File: neqo-crypto/min_version.rs
Comment (martinthomson) [inline neqo-crypto/min_version.rs:1]: See other comments, but this *might* be better off if it were moved to src/

File: .github/actions/nss/action.yml
Comment (martinthomson) [inline .github/actions/nss/action.yml:25]: For testing the availability of commands, I use `hash pkg-config 2>/dev/null`.  YMMV.  Note that just invoking pkg-config without testing for it first will also achieve the same:

```suggestion
```

**Review by martinthomson — APPROVED** [review]

**Comment by martinthomson** [comment]
OK, so approvals with comments get merged.  That's not ideal.

**Comment by martinthomson** [comment]
@larseggert should we require that conversations are resolved before merging?

image

**Comment by larseggert** [comment]
@martinthomson we can turn that on, but then we should probably turn **off** `Require approval of the most recent reviewable push`, because otherwise a re-review would be required just as it is now if one had chosen "request changes"?

File: neqo-crypto/src/lib.rs
Comment (larseggert) [reply neqo-crypto/src/lib.rs:62]: Do you have a preference in which directory `min_version.txt` and `min_version.rs` live? I wasn't sure whether to put one or both under `src`.

File: .github/actions/nss/action.yml
Comment (larseggert) [reply .github/actions/nss/action.yml:25]: Just invoking it if it isn't installed will fail with an error code != 0 and so stop CI.

**Comment by larseggert** [comment]
> I think that you can cut most of `fn nss_dir()` now that you won't be hitting it as often. Just pass the env variable in to `setup_standalone()`
 
@martinthomson: `nss_dir()` has the code that checks out NSS and NSPR if there is no copy already. Are you saying that should be cut, i.e., will we require people to do their own checkouts?

**Comment by martinthomson** [comment]
> turn off Require approval of the most recent reviewable push

Yeah, I think that we can trust people to exercise their judgment on the small stuff.

> will we require people to do their own checkouts?

Yeah.  You will observe that you can't reach that code unless you set NSS_DIR but haven't got a copy of NSS.  Given that the main reason to use NSS_DIR is to point to a copy you are working on, that seems like a good trade to me.

**Pull request merged**
